executor/executor.go

```diff
@@ -182,14 +182,19 @@ type Executor interface {
 
 // Next is a wrapper function on e.Next(), it handles some common codes.
 func Next(ctx context.Context, e Executor, req *chunk.RecordBatch) error {
-	sessVars := e.base().ctx.GetSessionVars()
+	base := e.base()
+	sessVars := base.ctx.GetSessionVars()
 	if atomic.CompareAndSwapUint32(&sessVars.Killed, 1, 0) {
 		return ErrQueryInterrupted
 	}
 	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
 		span1 := span.Tracer().StartSpan(fmt.Sprintf("%T.Next", e), opentracing.ChildOf(span.Context()))
 		defer span1.Finish()
 	}
+	if base.runtimeStats != nil {
```

Comment:
we should move it to the beginning of the function.

Reply:
What's the difference?

Reply:
We'd better cover the execution path as more as possible theoretically.